### DISCUSSION TITLE
Any way to use with angular 18 with SSR?

### DISCUSSION THREAD

**Discussion by Chistianxfp** [Leaflet/Leaflet#9412] [Q&A] [answered]
I I have been trying to apply Leaflet in many ways in my Angular 18 project with SSR, but I haven't found a way to make it work.
When importing it with "import * as L from 'leaflet';" it gives an error indicating that "window is not defined". I have tried to do it with services as shown in this link: https://stackoverflow.com/questions/58080301/how-to-use-angular-universal-with-leaflet but it throws this error: "Dynamic require of 'leaflet' is not supported". I have tried many things but haven't found a way to move forward. Any ideas?

**Comment by jonkoops** [comment]
See #9385

**Reply by Chistianxfp** [reply]
Still having the same problem when import L from leaflet and I dont want to disable SSR.
![error](https://github.com/user-attachments/assets/ac65d6a0-92fa-40be-804f-65134e472d10)

**Reply by jonkoops** [reply, marked as answer]
We don't have the aforementioned SSR enabler in a release right now, and even if we did we still won't support rendering a map on the server, but only avoid throwing errors such as the one you posted here. You'll have to wait until we do a release if you want those changes, which likely won't happen for a while.

You really should not be trying to run Leaflet in an SSR context, it is highly dependent on browser APIs (as are any plugins you might want to use), it is not made to be ran anywhere else, it will break. You should be able to tweak Angular Universal to not load Leaflet, this is a [well documented caveat](https://github.com/angular/universal/blob/main/docs/gotchas.md).

**Reply by Chistianxfp** [reply]
Thank you very much for your answers!
